Database migration

serdarsuezen opened the pull request:
Update of the oracle_to_exasol.sql database migration script.
The update contains script parameters that control the generation of primary and foreign keys, degree of parallelism for the data migration, checking tables for migration testing and some other improvements.

The option for parallel statements defines the degree of parallelism for the import statements. If a table is partitioned, the number of rows per partition will be determined to distribute the partitions across the parallel statements evenly. If the table does not have partitions, the ora_hash function on the row_id will be used to assign the rows to one of the parallel statements.

The script options for primary / foreign keys control the code being generated. The statements will be generated anyway. When the option is not set (FALSE) the statements will be commented out.

The script also offers the option to create checking tables. These tables are generated for every table in the scope. They will be created in the migrated schema. The name of the checking tables are the same as the orginal ones with the suffix "_MIG_CHK".
A summary table is created in the DATABASE_MIGRATION schema of the script with the name of the schema and the suffix "_MIG_CHK", allowing to query all metrics for all tables of the scope easily to confirm a complete, correct and successful migration.

serdarsuezen committed:
Update teradata_to_exasol.sql

removed check summary

serdarsuezen committed:
Update teradata_to_exasol.sql

added parameter for the script to generate the checking tables.
updated the checking table generation.

serdarsuezen committed:
Revert "Update teradata_to_exasol.sql"

This reverts commit ab543e2b8010d13feb23631082aa256f41d3fa12.

serdarsuezen committed:
Update teradata_to_exasol.sql

updated script

serdarsuezen committed:
Update teradata_to_exasol.sql

Removed duplicate with clause

serdarsuezen committed:
Update teradata_to_exasol.sql

serdarsuezen committed:
Update teradata_to_exasol.sql

serdarsuezen committed:
Update teradata_to_exasol.sql

serdarsuezen committed:
Update teradata_to_exasol.sql

Workaround for the character encoding error and output ordering

serdarsuezen committed:
Update teradata_to_exasol.sql

Adjusted join condition on the check summary

serdarsuezen committed:
Update teradata_to_exasol.sql

Adjusted/added comments

serdarsuezen committed:
Update teradata_to_exasol.sql

Adjusted comments

serdarsuezen committed:
Revert "Update teradata_to_exasol.sql"

This reverts commit 72e63680755cf51f96a3a2217802dc14ee4a028b.

serdarsuezen committed:
Update oracle_to_exasol.sql

Updte of the oracle to exasol database migration script.
The update contains script parameters that control the generation of primary and foreign keys, degree of parallelism for the data migration, checking tables for migration testing and some other improvements.

The option for parallel statements defines the degree of parallelism for the import statements. If a table is partitioned, the number of rows per partition will be determined to distribute the partions across the parallel statements evenly. If the table does not have partitions, the ora_hash function on the row_id will be used to assign the rows to one of the parallel statements.

The script options for primary / foreign keys control the code being generated. The statements will be generated anyway. When the option is not set (FALSE) the statements will be commented out.

The script also offers the option to create checking tables. These tables are generated for every table in the scope. They will be created in the migrated schema. The name of the checking tables are the same as the orginal ones with the suffix "_MIG_CHK".
A summary table is created in the database_migration schema of the script with the name of the schema and the suffix "_MIG_CHK", allowing to query all metric for all tables of the scope easily to confirm a complete, correct and successful migration.

exaValerieWiedemann reviewed (changes requested):
Can you please modify the example in Oracle_to_Exasol so that it migrates all tables in a given schema as this is the standard case for most first-time migrations? Thanks!

serdarsuezen committed:
Update oracle_to_exasol.sql

Changed the table filter condition from explicit table names, to all tables within a schema(s)

serdarsuezen committed:
Update oracle_to_exasol.sql

one % too much

exaValerieWiedemann committed:
Merge branch 'master' into database-migration
